src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -403,27 +419,39 @@ class DemuxFastqs
     val isUnmatched = sample.sampleName == UnmatchedSampleId
     val prefix = toSampleOutputPrefix(sample, isUnmatched, illuminaStandards, output, this.unmatched)
 
-    if (this.outputFastqs) {
-      new FastqRecordWriter(prefix, pairedEnd, illuminaStandards)
+    val writers = new ListBuffer[DemuxWriter]()
+
+    this.outputType match {
+      case OutputType.FastqOnly | OutputType.Both =>
+        writers += new FastqRecordWriter(prefix, pairedEnd, illuminaStandards)
+      case _ => Unit
     }
-    else {
-      val readGroup = new SAMReadGroupRecord(sample.sampleId)
-      readGroup.setSample(sample.sampleName)
-      readGroup.setLibrary(sample.libraryId)
-      readGroup.setPlatform("Illumina")
-      sample.description.foreach(readGroup.setDescription)
-      platformUnit.foreach(readGroup.setPlatformUnit)
-      sequencingCenter.foreach(readGroup.setSequencingCenter)
-      predictedInsertSize.foreach(readGroup.setPredictedMedianInsertSize)
-      runDate.foreach(readGroup.setRunDate)
-      platformModel.foreach(readGroup.setPlatformModel)
-
-      val header: SAMFileHeader = new SAMFileHeader
-      header.addReadGroup(readGroup)
-      header.setSortOrder(sortOrder)
-      comments.foreach(header.addComment)
-
-      new SamRecordWriter(PathUtil.pathTo(prefix + ".bam"), header, this.umiTag, numSamples)
+
+    this.outputType match {
+      case OutputType.BamOnly | OutputType.Both =>
+          val readGroup = new SAMReadGroupRecord(sample.sampleId)
+          readGroup.setSample(sample.sampleName)
+          readGroup.setLibrary(sample.libraryId)
+          readGroup.setPlatform("Illumina")
+          sample.description.foreach(readGroup.setDescription)
+          platformUnit.foreach(readGroup.setPlatformUnit)
+          sequencingCenter.foreach(readGroup.setSequencingCenter)
+          predictedInsertSize.foreach(readGroup.setPredictedMedianInsertSize)
+          runDate.foreach(readGroup.setRunDate)
+          platformModel.foreach(readGroup.setPlatformModel)
+
+          val header: SAMFileHeader = new SAMFileHeader
+          header.addReadGroup(readGroup)
+          header.setSortOrder(sortOrder)
+          comments.foreach(header.addComment)
+
+          writers += new SamRecordWriter(prefix, header, this.umiTag, numSamples)
+      case _ => Unit
+    }
+
+    new DemuxWriter {
```

Comment:
I think the answer is no, but do we need to worry about memory requirements when writing both?  Or is the case that BAM sorts and fastq doesn't, so it's not double?